arelle/plugin/validate/HMRC/__init__.py

```diff
@@ -166,16 +166,19 @@ def validateXbrlStart(val, parameters=None, *args, **kwargs):
         val.isAccounts = not val.isComputation
 
     val.txmyType = None
-    for doc in val.modelXbrl.modelDocument.referencesDocument:
-        ns = doc.targetNamespace
-        if ns:
-            if ns.startswith("http://www.xbrl.org/uk/char/"): val.txmyType = "charities"
-            elif ns.startswith("http://www.xbrl.org/uk/gaap/"): val.txmyType = "ukGAAP"
-            elif ns.startswith("http://www.xbrl.org/uk/ifrs/"): val.txmyType = "ukIFRS"
-            elif FRC_2022_NAMESPACE.search(ns): val.txmyType = "FRS-2022"
-            elif ns.startswith("http://xbrl.frc.org.uk/"): val.txmyType = "FRS"
-            else: continue
-            break
+    if any((concept.qname.namespaceURI.startswith(FRC_URL_DOMAIN) and concept.modelDocument.inDTS)
+           for concept in val.modelXbrl.nameConcepts.get("AccountsType", ())):
+        val.txmyType = "FRS-2022"
+    else:
+        for doc in val.modelXbrl.modelDocument.referencesDocument:
+            ns = doc.targetNamespace
+            if ns:
+                if ns.startswith("http://www.xbrl.org/uk/char/"): val.txmyType = "charities"
+                elif ns.startswith("http://www.xbrl.org/uk/gaap/"): val.txmyType = "ukGAAP"
+                elif ns.startswith("http://www.xbrl.org/uk/ifrs/"): val.txmyType = "ukIFRS"
+                elif ns.startswith(FRC_URL_DOMAIN): val.txmyType = "FRS"
+                else: continue
```

Comment:
This is fine and pre-existing, but probably way easier to understand with something like:

```python
def get_txmy_type(referencesDocument) -> str | None:
    ns_prefix_to_txmy_type =  {
        'http://www.xbrl.org/uk/char/': 'charities',
        'http://www.xbr.org.uk.gaap/': 'ukGAAP',
        'http://www.xbrl.org.uk/ifrs/': 'ukIFRS',
        FRC_URL_DOMAIN: 'FRC'
    }
    for doc in referencesDocument:
        doc_ns = doc.targetNamespace
        if not doc_ns:
            continue
        for ns_prefix, txmy_type in ns_prefix_to_txmy_type.items():
            if doc_ns.startswith(ns_prefix):
                return txmy_type
    return None
   
```